java/arcs/core/analysis/ExpressionClaimDeducer.kt

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+package arcs.core.analysis
+
+import arcs.core.data.Claim
+import arcs.core.data.expression.Expression
+
+/**
+ * A visitor to help create [Claim]s for Paxel [Expression]s.
+ *
+ * For each [Expression], this visitor produces a [Deduction], which can be translated into a set
+ * of [Claim] relationships.
+ *
+ * [Deduction]s collect [Expression.FieldExpression]s as [Deduction.Equal]s and associate
+ * them with other fields. These are recursive structures, and can represent [Claim] relationships
+ * for deeply nested [Expression]s.
+ */
+class ExpressionClaimDeducer : Expression.Visitor<Deduction, Unit> {
```

Comment:
Arguably we don't produce Claims here. So maybe `ExpressionDataFlowAnalyzer` ? Or simply `ExpressionAnalyzer` ?

Then we will have a component that takes the result of the analysis and turns them into claims.

WDYT?

Reply:
Yes, I am on the same page. I like `ExpressionAnalyzer`.

Reply:
Oh, I said `ExpressionClaimAnalyzer` above. That's another option.